src/realm_dart.cpp

```diff
@@ -92,63 +86,15 @@ RLM_API void realm_dart_delete_persistent_handle(void* handle) {
     Dart_DeletePersistentHandle_DL(persistentHandle);
 }
 
-namespace {
-
-using FreeT = std::function<void()>;
-using CallbackT = std::function<void(const realm_http_request_t request, void* request_context)>; // Differs per callback
-using UserdataT = std::tuple<realm_scheduler_t*, CallbackT, FreeT>;
-
-
-void _callback(realm_userdata_t userdata, const realm_http_request_t request, void* request_context) {
-    // the pointers in error are to stack values, we need to make copies and move them into the scheduler invocation
-    struct request_copy_buf {
-        std::string url;
-        std::string body;
-        std::map<std::string, std::string> headers;
-        std::vector<realm_http_header_t> headers_vector;
-    } buf;
-
-    buf.url = request.url;
-    buf.body = std::string(request.body, request.body_size);
-    for (size_t i = 0; i < request.num_headers; i++) {
-        auto [it, _] = buf.headers.emplace(request.headers[i].name, request.headers[i].value);
-        buf.headers_vector.push_back({ it->first.c_str(), it->second.c_str() });
-    }
-
-    realm_http_request_t request_copy = request; // copy struct
-    request_copy.url = buf.url.c_str();
-    request_copy.body = buf.body.data();
-    request_copy.headers = buf.headers_vector.data();
-
-    // trampoline to scheduler
-    auto u = reinterpret_cast<UserdataT*>(userdata);
-    auto scheduler = std::get<0>(*u);
-    auto callback = std::get<1>(*u);
-    (*scheduler)->invoke([buf = std::move(buf), request = std::move(request_copy), callback = std::move(callback), request_context]{
-        callback(request, request_context);
-    });
-}
-
-void _userdata_free(void* userdata) {
-    auto u = reinterpret_cast<UserdataT*>(userdata);
-    auto scheduler = std::get<0>(*u);
-    auto userdata_free = std::get<2>(*u);
-    (*scheduler)->invoke([userdata_free = std::move(userdata_free), u]{
-        userdata_free();
-        delete u;
-    });
+RLM_API realm_dart_userdata_async_t realm_dart_userdata_async_new(Dart_Handle handle, void* callback, realm_scheduler_t* scheduler)
+{
+    return new realm_dart_userdata_async(handle, callback, scheduler);
 }
```

Comment:
Why not create this in Dart by exposing the realm_dart_userdata_async struct

Reply:
this will save us one native transition. But also callback registration happens once per registration so its not on a hot path.

Reply:
`realm_dart_userdata_async_t` is a struct with C++ members and cannot be easily allocated from Dart. What's more, it has a `Dart_PersistentHandle` member which we can't set from Dart - FFI only supports `Dart_Handle` types as parameters or return values of function calls. That's why this has to be constructed in native.

Reply:
I guess we were doing two transitions before to create the persistence handles so resolving this comment